### PR TITLE
Add documentation for -XX:[+|-]HandleSIGUSR2

### PR DESCRIPTION
**Pull request by babsingh**
Related: eclipse-openj9/openj9#16974

### CODE REVIEW

**Comment by pshipton** [comment]
The release notes also need to be updated. I expect to put the change in 0.38.

File: docs/xxhandlesigusr2.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+| Setting               | Effect  | Default                                                                            |
+|-----------------------|---------|:----------------------------------------------------------------------------------:|
+| `-XX:+HandleSIGUSR2 ` | Enable  | :fontawesome-solid-check:{: .yes aria-hidden="true"}<span class="sr-only">yes</span> |
+| `-XX:-HandleSIGUSR2 ` | Disable |                                                                                    |
+
+
+## Explanation
+
+When enabled, the VM handles the signal `SIGUSR2` and generates the various dump files.
```
Comment (pshipton) [inline docs/xxhandlesigusr2.md:43]: https://www.eclipse.org/openj9/docs/version0.35/#new-user2-event-added-for-the-xdump-option referenced the -Xdump and [Signal handling](https://www.eclipse.org/openj9/docs/openj9_signals/) doc, we could reference the same places.

Comment (pshipton) [reply docs/xxhandlesigusr2.md:43]: This doc should mention the `user2` event, and the doc for the usr2 event should reference this option.

**Comment by doveye** [comment]
@babsingh Can you please make the following changes:
- Please open an issue and fill in the template fields, assign it to yourself, then connect this PR to it. (Feel free to assign the issue to Sreekala if you'd rather we made the updates :slightly_smiling_face:)
- Please change the copyright dates to match the other topics - there is only one set of dates for the docs, as described in the [contribution guidelines](https://github.com/eclipse-openj9/openj9-docs/blob/master/CONTRIBUTING.md)
- An entry needs to be added to the mkdocs.yml file to make the new topic appear in the table of contents
- The version0.38.md file (What's New) will need to be updated when it's created (currently, the docs are still being updated for 0.37)
- The [-Xrs topic](https://www.eclipse.org/openj9/docs/xrs/) needs updating, eg the new option should be added to the note (I wonder if -XX:+HandleSIGXFSZ should be added too?) And should SIGUSR2 (and SIGXFSZ) be added to the list after "If you specify the sync parameter"?
- The [Signal handling](https://www.eclipse.org/openj9/docs/openj9_signals/) topic also needs updating, eg to add the new option in the note where -XX[+|-]HandleSIGABRT is mentioned (-XX:+HandleSIGXFSZ probably needs adding here too - I guess the doc updates for that weren't as thorough as for HandleSIGABRT). Also, does -Xrs:sync need to be added as a disable option to the table rows for SIGUSR2? (I see that the code update contains this comment: "Throw error message if both -XX:+HandleSIGUSR2 and -Xrs/-Xrs:sync are supplied.")

Does this update have anything to do with HotSpot? Just wondering because HandleSIGABRT is mentioned in https://www.eclipse.org/openj9/docs/cmdline_migration/#creating-compatible-behavior.

**Comment by babsingh** [comment]
I didn't know that the requirements had changed. Last time, I remember creating the docs PR myself. I have created https://github.com/eclipse-openj9/openj9-docs/issues/1077 and assigned it to @Sreekala-Gopakumar.

**Comment by babsingh** [comment]
> The [-Xrs topic](https://www.eclipse.org/openj9/docs/xrs/) needs updating, eg the new option should be added to the note (I wonder if -XX:+HandleSIGXFSZ should be added too?) And should SIGUSR2 (and SIGXFSZ) be added to the list after "If you specify the sync parameter"?

No for HandleSIGXFSZ. Xrs and HandleSIGXFSZ can be provided together. Such error handling does not exist for HandleSIGXFSZ at this point.

No for adding to the sync list. SIGUSR2 is an async signal. `-Xrs:async` is not formally documented.

> does -Xrs:sync need to be added as a disable option to the table rows for SIGUSR2? (I see that the code update contains this comment: "Throw error message if both -XX:+HandleSIGUSR2 and -Xrs/-Xrs:sync are supplied.")

No. Only `-Xrs` or `-Xrs:async` will disable the new SIGUSR2 option. I have updated https://github.com/eclipse-openj9/openj9/pull/16974 in regards to this. `-Xrs:async` is not documented. Documentation should only state: `Throw error message if both -XX:+HandleSIGUSR2 and -Xrs are supplied.`

> Does this update have anything to do with HotSpot? 

No.

**Comment by doveye** [comment]
Thanks @babsingh. You can still create PRs if you want, but they do need an issue and if you assign it to yourself, we'll know not to pick it up 🙂

**Comment by babsingh** [comment]
Replaced with https://github.com/eclipse-openj9/openj9-docs/pull/1080.

**Pull request closed without merging**